tests/upgrade-tests/utils.py

```diff
@@ -45,26 +46,45 @@ def dump_model_info(model):
     return model
 
 
+@contextmanager
+def timeout_for_current_task(timeout):
+    ''' Create a context with a timeout.
+
+    If the context body does not finish within the time limit, then the current
+    asyncio task will be cancelled, and an asyncio.TimeoutError will be raised.
+    '''
+    loop = asyncio.get_event_loop()
+    task = asyncio.Task.current_task()
+    handle = loop.call_later(timeout, task.cancel)
+    try:
+        yield
+    except asyncio.CancelledError:
+        raise asyncio.TimeoutError('Timed out after %f seconds' % timeout)
+    finally:
+        handle.cancel()
```

Comment:
This approach seems weird but I couldn't come up with anything better. Open to suggestions!